### DISCUSSION TITLE
What is the API to get multiple fields with the same name?

### DISCUSSION THREAD

**Discussion by XVilka** [tree-sitter/tree-sitter#1382] [Q&A] [answered]
Lets take the example of the [C grammar](https://github.com/tree-sitter/tree-sitter-c):
```c
struct bla { int foo, bar; };
```
produces the following tree:
```
translation_unit [0, 0] - [1, 0]
  struct_specifier [0, 0] - [0, 28]
    name: type_identifier [0, 7] - [0, 10]
    body: field_declaration_list [0, 11] - [0, 28]
      field_declaration [0, 13] - [0, 26]
        type: primitive_type [0, 13] - [0, 16]
        declarator: field_identifier [0, 17] - [0, 20]
        declarator: field_identifier [0, 22] - [0, 25]
```

I usually use code like this:
```c
TSNode field_type = ts_node_child_by_field_name(child, "type", 4);
TSNode field_declarator = ts_node_child_by_field_name(child, "declarator", 10);
```
Is there any suitable API to walk over the multiple nodes named "declarator"? I found nothing similar in the documentation.

**Comment by maxbrunsfeld** [comment, marked as answer]
Yeah, the more general way of accessing fields is via the Tree cursor API. You can efficiently walk a tree with a cursor, and at every step of the walk, you can find out the field name for the current node using [`ts_tree_cursor_current_field_name`](https://github.com/tree-sitter/tree-sitter/blob/22a5cfbe102056ff3d51b2b44a8666fd8f85fdc0/lib/include/tree_sitter/api.h#L613-L619).
